sdks/java/extensions/avro/src/main/java/org/apache/beam/sdk/extensions/avro/schemas/utils/AvroUtils.java

```diff
@@ -1022,9 +1046,23 @@ private static org.apache.avro.Schema getFieldSchema(
         break;
 
       case DECIMAL:
-        baseType =
-            LogicalTypes.decimal(Integer.MAX_VALUE)
-                .addToSchema(org.apache.avro.Schema.create(Type.BYTES));
+        // Beam avro decimal defaults: type=bytes, precision=MAX_INT, scale=0
+        String type = options.hasOption("type") ? options.getValue("type") : "bytes";
```

Comment:
I think it somehow should work via explicit LogicalType rather than untyped options.